packages/web3-providers-http/src/index.js

```diff
@@ -46,6 +46,7 @@ var HttpProvider = function HttpProvider(host, options) {
     }
     this.timeout = options.timeout || 0;
     this.headers = options.headers;
+    this.agent = options.agent || this.agent;
```

Comment:
Thanks for opening this PR! This line doesn't get used below and the method ``nodeJsSet()`` does accept the following options object:

``` typescript
interface {
  httpAgent: Agent;
  httpsAgent: Agent;
  baseUrl: string;
}
``` 

Could you implement it in a way to have the possibility to pass all ``nodejsSet`` related option object properties with respecting of the automatic detection of the correct agent on line [42](https://github.com/ethereum/web3.js/blob/fc76d4a22f745d84c13767a326437b9a81f00044/packages/web3-providers-http/src/index.js#L42)?

Reply:
Thanks for the review! I will adjust the PR asap.